controller/tests/server/update/outcome.ml

```diff
@@ -25,20 +25,21 @@ let slot_spec_to_outcome ({booted_slot; primary_slot; input_versions} : Helpers.
 (* Checks if the state returned by UpdateService matches
    the expected outcome as determined by [slot_spec_to_outcome] *)
 let state_matches_expected_outcome state outcome =
-    match (outcome, state) with
-        | (InstallVsn v1,             Update.Downloading v2) ->
-                (Semver.to_string v1) = v2
-        | (InstallVsn _,              _) ->                         false
-        | (DoNothingOrProduceWarning, Update.ErrorGettingVersionInfo _) -> true
-        | (DoNothingOrProduceWarning, Update.UpToDate _) ->                true
-        | (DoNothingOrProduceWarning, Update.OutOfDateVersionSelected) ->  true
-        | (DoNothingOrProduceWarning, Update.RebootRequired) ->            true
-        | (DoNothingOrProduceWarning, Update.ReinstallRequired) ->         true
-        (* should not _directly_ return to GettingVersionInfo state *)
-        | (DoNothingOrProduceWarning, Update.GettingVersionInfo) ->        false
-        (* all the other states are part of the installation process
-           and are treated as errors *)
-        | (DoNothingOrProduceWarning, _) ->                         false
+    let open Update in
+    match (outcome, state.system_status, state.process_state) with
```

Comment:
Note: the spec here is made stricter by verifying that the returned `version_info` is updated and that the process is in the expected `process_state`, but otherwise is not changed.